src/Umbraco.Infrastructure/PropertyEditors/MediaPicker3PropertyEditor.cs

```diff
@@ -188,48 +204,63 @@ internal static IEnumerable<MediaWithCropsDto> Deserialize(IJsonSerializer jsonS
             }
         }
 
-        // private JArray PersistTempMedia(JArray jArray, MediaPicker3Configuration mediaPicker3Configuration)
-        // {
-        //     var result = new JArray();
-        //     foreach (JObject? dto in jArray.Values<JObject>())
-        //     {
-        //         if (dto is null)
-        //         {
-        //             continue;
-        //         }
-        //
-        //         if (!dto.TryGetValue("tmpLocation", out JToken? temporaryLocation))
-        //         {
-        //             // If it does not have a temporary path, it can be an already saved image or not-yet uploaded temp-image, check for media-key
-        //             if (dto.TryGetValue("mediaKey", out _))
-        //             {
-        //                 result.Add(dto);
-        //             }
-        //
-        //             continue;
-        //         }
-        //
-        //         var temporaryLocationString = temporaryLocation.Value<string>();
-        //         if (temporaryLocationString is null)
-        //         {
-        //             continue;
-        //         }
-        //
-        //         GuidUdi? startNodeGuid = mediaPicker3Configuration.StartNodeId as GuidUdi ?? null;
-        //         JToken? mediaTypeAlias = dto.GetValue("mediaTypeAlias");
-        //         IMedia mediaFile = _temporaryMediaService.Save(temporaryLocationString, startNodeGuid?.Guid, mediaTypeAlias?.Value<string>());
-        //         MediaWithCropsDto? mediaDto = _jsonSerializer.Deserialize<MediaWithCropsDto>(dto.ToString());
-        //         if (mediaDto is null)
-        //         {
-        //             continue;
-        //         }
-        //
-        //         mediaDto.MediaKey = mediaFile.GetUdi().Guid;
-        //         result.Add(JObject.Parse(_jsonSerializer.Serialize(mediaDto)));
-        //     }
-        //
-        //     return result;
-        // }
+        private List<MediaWithCropsDto> UpdateMediaTypeAliases(List<MediaWithCropsDto> mediaWithCropsDtos)
+        {
+            const string unknownMediaType = "UNKNOWN";
+
+            foreach (MediaWithCropsDto mediaWithCropsDto in mediaWithCropsDtos)
+            {
+                IMedia? media = _mediaService.GetById(mediaWithCropsDto.MediaKey);
+                mediaWithCropsDto.MediaTypeAlias = media?.ContentType.Alias ?? unknownMediaType;
+            }
+
+            return mediaWithCropsDtos.Where(m => m.MediaTypeAlias != unknownMediaType).ToList();
+        }
+
+        List<MediaWithCropsDto> HandleTemporaryMediaUploads(List<MediaWithCropsDto> mediaWithCropsDtos, MediaPicker3Configuration configuration)
```

Comment:
```suggestion
        private List<MediaWithCropsDto> HandleTemporaryMediaUploads(List<MediaWithCropsDto> mediaWithCropsDtos, MediaPicker3Configuration configuration)
```

I'm assuming we want this to be private 😄